### DISCUSSION TITLE
multi-objective  optimization ( urgent help needed)

### DISCUSSION THREAD

**Discussion by hosseinnegahban** [mdolab/dafoam#313] [Tutorials]
Hi,

I have 2 objective functions with one constraint function and I want to minimize one of the objectives and maximize the other and finally divide the first to the second one. I have used channel tutorial for this and I added objectives, and it solves, but the problem is  both of them are minimized and the sum of them is extracted as objective value. How can I minimize one and maximize the other and divide the final result    (max obj1 & min obj2   -------> obj1/obj2)

It is really urgent, I would appreciate if you kindly help me with this problem.

Thank you very much.

**Comment by friedenhe** [comment]
You can assign a `-1` scaling to an objective function such that it will be maximized during the optimization.

**Comment by hosseinnegahban** [comment]
Thank you for your reply. Yes i put -1 to the one i want to max and 1 to the one that i wnat to minimize but how could i divide the obj1 to obj2 in every iteration?
________________________________

**Comment by friedenhe** [comment]
They will be printed on the screen when the primal solves...

**Reply by hosseinnegahban** [reply]
To be more precise, I want to maximize Cl/Cd but could I use them as two objective functions and max Cl and min Cd and then divide manually, or I have to define Cl/Cd as one objective function?

Thank you in advance for your reply.

**Comment by friedenhe** [comment]
Check this script: https://github.com/DAFoam/tutorials/blob/ed353b90b59c563d698f3dd7e5b9a22340ea7acd/NACA0012_Airfoil/incompressible/runScript_LOD.py

**Reply by hosseinnegahban** [reply]
Professor,

Thank you again for your great workshop today. 

I have a question regarding the code u sent me regarding lift to drag ratio as objective function. I have run my case with this code, but the problem is it is not maximizing instead it is minimizing  I also checked "scale" it is -1. I would appreciate it if you help me in this regard. I have attached the run script file as well as IPOP file in zip file for your consideration.
[Desktop.zip](https://github.com/mdolab/dafoam/files/8865195/Desktop.zip)

**Reply by friedenhe** [reply]
We just added this case script back to the tutorial. Can you run this script (without any change) using Docker? https://github.com/DAFoam/tutorials/blob/main/NACA0012_Airfoil/incompressible/runScript_LoD.py

We managed to increase L/D without a problem.

**Reply by hosseinnegahban** [reply]
Thank you very much for your reply and for the new code. I ran this code on naca0012 incompressible and it maximized Cl/Cd, but at the end it showed the message "EXIT: Restoration Failed!" after 15 iterations. I have attached it IPOPT file for your consideration. What does this error mean?
[opt_IPOPT.txt](https://github.com/mdolab/dafoam/files/8867134/opt_IPOPT.txt)

**Comment by hosseinnegahban** [comment]
Thank you very much . I will test it as u recommend, im using dafoam with source code is it ok not to use docker for this problem?
________________________________
